### PR TITLE
Mitigate `noAccountError` #2098678

### PR DESCRIPTION
**Pull request by konstantin-msft**
- Log number of accounts in trace mode.
- Validate that `getActiveAccount()` picks up legacy account identifier from local storage.

### CODE REVIEW

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts [hunk 1]
```diff
@@ -888,7 +935,7 @@ describe("PublicClientApplication.ts Class Unit Tests", () => {
                 expect(request.scopes).toContain("openid");
                 expect(request.scopes).toContain("profile");
                 done();
-                
+
```
Comment (bmahall) [inline lib/msal-browser/test/app/PublicClientApplication.spec.ts:948]: nit: Formatting changes generally to be part of a separate PR.

File: change/@azure-msal-browser-05d052ac-925b-40f2-8d36-78bcab460285.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "- Log number of accounts in trace mode. - Validate that `getActiveAccount()` picks up legacy account identifier from local storage.",
```
Comment (bmahall) [inline change/@azure-msal-browser-05d052ac-925b-40f2-8d36-78bcab460285.json]:
```suggestion
  "comment": "- Log number of accounts in trace mode. - Validate that `getActiveAccount()` picks up legacy account identifier from local storage. #5525",
```

**Review by bmahall — APPROVED** [review]

File: change/@azure-msal-browser-05d052ac-925b-40f2-8d36-78bcab460285.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "- Log number of accounts in trace mode. - Validate that `getActiveAccount()` picks up legacy account identifier from local storage. #5525",
```
Comment (tnorling) [inline change/@azure-msal-browser-05d052ac-925b-40f2-8d36-78bcab460285.json]:
```suggestion
  "comment": "Log number of accounts in trace mode. #5525",
```

Comment (tnorling) [reply change/@azure-msal-browser-05d052ac-925b-40f2-8d36-78bcab460285.json]: I don't see any changes to the lib besides some logger changes?

**Review by tnorling — APPROVED** [review]

File: change/@azure-msal-browser-05d052ac-925b-40f2-8d36-78bcab460285.json (hunk 3 above)
Comment (konstantin-msft) [reply change/@azure-msal-browser-05d052ac-925b-40f2-8d36-78bcab460285.json]: There is also a new unit test here https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5525/files#diff-f936803be34376b4fc36745b78ec5110b6e3b420d907b033fa6924453d2815d5R3235 to make sure old account ids from local cache are honored.

Comment (tnorling) [reply change/@azure-msal-browser-05d052ac-925b-40f2-8d36-78bcab460285.json]: Gotcha. Since this will be published in our release notes we should stick to just what has changed in the lib
